code/datums/traits/good.dm

```diff
@@ -219,3 +219,9 @@
 /datum/quirk/night_vision/on_spawn()
 	var/mob/living/carbon/human/H = quirk_holder
 	H.update_sight()
+
+/datum/quirk/hardsoles
+	name = "Hardened Soles"
+	desc = "The soles of your feet are hardened from walking barefoot for so long. You no longer have to worry about stepping on anything, sharp or otherwise."
```

Comment:
Guess you'll have to edit the description a little after applying the suggestions.